File: app/src/main/java/vergecurrency/vergewallet/service/model/wallet/InputOutput.java
package vergecurrency.vergewallet.service.model.wallet;

import com.google.gson.Gson;

import vergecurrency.vergewallet.Constants;

public class InputOutput {

	private int amount;
	private String address;
	private boolean isMine;

	public static InputOutput decode(String JSON) {
		return new Gson().fromJson(JSON, InputOutput.class);
	}

	public float amountValue() {
		return (float) (amount / Constants.SATOSHIS_DIVIDER);
	}

	public int getAmount() {
		return amount;
	}

	public void setAmount(int amount) {
		this.amount = amount;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public boolean isMine() {
		return isMine;
	}

	public void setMine(boolean mine) {
		isMine = mine;
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/service/model/wallet/InputOutput.kt
package vergecurrency.vergewallet.service.model.wallet

import com.google.gson.Gson

import vergecurrency.vergewallet.Constants

class InputOutput {

    var amount: Int = 0
    var address: String? = null
    var isMine: Boolean = false

    fun amountValue(): Float {
        return (amount / Constants.SATOSHIS_DIVIDER).toFloat()
    }

    companion object {

        fun decode(JSON: String): InputOutput {
            return Gson().fromJson(JSON, InputOutput::class.java)
        }
    }
}
